proxy/mux/mux.go

```diff
@@ -175,6 +175,7 @@ func (m *mux) Init(ss engine.Snapshot) error {
 		m.servers[lsnCfg.Key()] = srv
 	}
 
+	routes := make(map[string]interface{})
```

Comment:
Small nitpick. Can you initialize with `len(ss.FrontendSpecs)`? Especially since this optimization is for 1600 routes, that's 1600 slice resizes in the worst case.